Jenkinsfile

```diff
@@ -13,19 +13,9 @@ pipeline {
                 sh 'flake8 --select=DUO > reports/dlint.txt'
             }
         }
-        stage('Build') {
+         stage('Tests') {
             steps {
-                sh 'pip install .'
-                sh 'pip install freezegun'
-                sh 'pip install validators'
-                sh 'pip install ukpostcodeparser'
-                sh 'pip install random2'
-                sh 'pip install pytest'
-            }
-        }
-        stage('Tests') {
-            steps {
-                sh 'py.test -rA > reports/tests.txt'
+                sh 'python setup.py test > reports/tests.txt'
```

Comment:
You can use two steps: `tests: build environment` and `tests: run`.  You can do a pipeline for tests and execute more atomic and logic steps